### DISCUSSION TITLE
Question about force close and negative blocks_til_maturity

### DISCUSSION THREAD

**Discussion by LNOrville** [lightningnetwork/lnd#7684] [Q&A] [answered]
I had another involuntary force close. Channel was opened by my peer. I had 124,454 on my side, of which 47,587 was in an outgoing HTLC at the time of closing.
Now my node is trying to sweep funds. Out of the 76867 left in the channel I have received 18,731, the rest went to pay the 439 sat/vB. 

lncli pendingchannels says:

 "closing_txid": "fd6186c35f76247ca7c0c68979a135185be930cc5e6fcdf75951d772ba0d80ab",
            "limbo_balance": "12874",
            "maturity_height": 789097,
            "blocks_til_maturity": -22,
            "recovered_balance": "76867",
            "pending_htlcs": [
                {
                    "incoming": false,
                    "amount": "12874",
                    "outpoint": "9c746266450870bde2c6c8a2a999d9f2807b321d84bed015cfc183b9f358f5d7:0",
                    "maturity_height": 789098,
                    "blocks_til_maturity": -21,
                    "stage": 2
                }
            ],
            "anchor": "LIMBO"

We are now at block 789125. The 12874 is the sorry remains of my 47587 HTLC - but that one got swept in back in block 788954 (tx 9c746266450870bde2c6c8a2a999d9f2807b321d84bed015cfc183b9f358f5d7) - so why is this force close still pending? I did not receive the 12874 into my on-chain wallet. Will there be more expensive sweeping to do?

**Comment by guggero** [comment, marked as answer]
The sweeper is attempting to sweep the HTLC at an economic rate. And since it's "only" 12874 sats, with the current fee rates that wasn't possible. The -21 blocks just means it's been trying for 21 blocks and will continue to try. With the fee rates coming down it should probably happen soon.

**Comment by LNOrville** [comment]
Thank you for the explanation :-)
Bonus question: Is there a way to specify that sweeps of 330 sats anchors should not be performed at high tx fees or automatically bumped by CPFP, so situations like [this](https://mempool.space/tx/e13ad2ad69ec96caa3871b074d1340faabd2c556edf637ed8eca523fdde5289c) can be avoided? Not my FC, but it is scary.

**Reply by guggero** [reply]
Confirming force-close transactions in a timely manner can be safety critical, since there might be HTLCs that need to be resolved on chain before a certain block height. But I guess if there are no HTLCs then the sweeper could use lower fee rates. There's some discussion going on here: https://github.com/lightningnetwork/lnd/issues/7602.

**Reply by LNOrville** [reply]
Thanks for the link over to the other discussion. Very interesting!
